homeassistant/components/voice_assistant/websocket_api.py

```diff
@@ -46,38 +57,84 @@ async def websocket_run(
 
     else:
         # Construct a pipeline for the required/configured language
-        language = msg.get("language", hass.config.language)
         pipeline = Pipeline(
             name=language,
             language=language,
-            conversation_engine=None,
-            tts_engine=None,
+            stt_engine=None,  # cloud
+            conversation_engine=None,  # default agent
+            tts_engine=None,  # cloud
         )
 
-    # Run pipeline with a timeout.
-    # Events are sent over the websocket connection.
     timeout = msg.get("timeout", DEFAULT_TIMEOUT)
-    run_task = hass.async_create_task(
-        TextPipelineRequest(
-            intent_input=msg["intent_input"],
-            conversation_id=msg.get("conversation_id"),
-        ).execute(
-            PipelineRun(
-                hass,
-                connection.context(msg),
-                pipeline,
-                event_callback=lambda event: connection.send_event(
-                    msg["id"], event.as_dict()
+    ws_result: Any | None = None
+
+    intent_input = msg.get("intent_input")
+    if intent_input is None:
```

Comment:
I wonder if we should be more explicit in requesting the type of run

Reply:
Using stages instead